Ports/JavaSE/src/com/codename1/impl/javase/LocationSimulation.java

```diff
@@ -64,170 +79,278 @@ public LocationSimulation() {
+//                + "}\n"
+//                + "function initialize() {\n"
+//                + "var latlng = new google.maps.LatLng(40.714353, -74.005973 );\n"
+//                + "var myOptions = {\n"
+//                + "  zoom: "+startingZoom+",\n"
+//                + "  center: latlng,\n"
+//                + "  mapTypeControl: true,\n"
+//                + "  navigationControl: true,\n"
+//                + "  streetViewControl: true,\n"
+//                + "  backgroundColor: \"#FFFFFF\"\n"
+//                + "};\n"
+//                + "\n"
+//                + "document.geocoder = new google.maps.Geocoder();\n"
+//                + "document.map = new google.maps.Map(document.getElementById(\"map_canvas\"),myOptions);\n"
+//                + "console.log('map', document.map);\n"
+//                + "\n"
+//                + "document.marker = new google.maps.Marker({\n"
+//                + "    position: document.map.getCenter(),\n"
+//                + "    icon: {\n"
+//                + "      path: google.maps.SymbolPath.CIRCLE,\n"
+//                + "      scale: 5\n"
+//                + "    },\n"
+//                + "    map: document.map\n"
+//                + "  });"
+//                + "google.maps.event.addListener(document.map, 'drag', function() { document.marker.setPosition(document.map.getCenter()); } );"
+//                + "}"
+//                + "document.updateJavaFX = function updateJavaFX() {\n"
+//                + "    document.currentCenter  = document.map.getCenter();\n"
+//                + "    document.currentBounds  = document.map.getBounds();\n"
+//                + "    document.currentHeading = document.map.getHeading();\n"
+//                + "    document.currentZoom    = document.map.getZoom();\n"
+//                + "    document.marker.setPosition(document.currentCenter);\n"
+//                + "}"
+//                + "    </script>\n"
+//                + "  </head>\n"
+//                + " <body onload=\"initialize()\">\n"
+//                + "    <div id=\"map_canvas\" style=\"width:100%; height:100%\"></div>\n"
+//                + " </body>"
+//                + "</html>";
+
+        final JFXPanel webContainer = new JFXPanel();
+        mapPanel.setLayout(new BorderLayout());
+        mapPanel.add(BorderLayout.CENTER, webContainer);
+
+        Platform.runLater(() -> {
+            StackPane root = new StackPane();
+            WebView webView = new WebView();
+            WebEngine webEngine = webView.getEngine();
+            this.webView = webView;
+            this.webEngine = webEngine;
+            root.getChildren().add(webView);
+            webContainer.setScene(new Scene(root));
+
+            // Load the HTML content
+            webEngine.loadContent(htmlpage2, "text/html");
+
+            // Add a listener for the load state
+            webEngine.getLoadWorker().stateProperty().addListener((obs, oldState, newState) -> {
```

Comment:
This code needs to be Java 7 compliant due to historical reasons and unfortunately we can't use Lambdas. We need to use inner classes. See the CI compilation error here: https://github.com/codenameone/CodenameOne/actions/runs/12897154590/job/35968885875?pr=3863